node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -921,13 +924,58 @@ public Mutex lock(ApplicationId application, Duration timeout) {
     /** Create a lock which provides exclusive rights to modifying unallocated nodes */
     public Mutex lockUnallocated() { return db.lockInactive(); }
 
-    /** Acquires the appropriate lock for this node */
-    public Mutex lock(Node node) {
+    /** Returns the unallocated/application lock, and the node acquired under that lock. */
+    public Optional<NodeMutex> lockAndGet(Node node) {
+        Node staleNode = node;
+
+        for (int i = 0; i < 4; ++i) {
```

Comment:
Consider extracting retries as a constant, and include it in the exception message.

Reply:
Done